tests/trainer/logging/test_train_loop_logging_1_0.py

```diff
@@ -489,3 +477,70 @@ def validation_step(self, batch, batch_idx):
         weights_summary=None,
     )
     trainer.fit(model, train_data, val_data)
+
+
+def test_misconfiguration_error_for_training_step(tmpdir):
```

Comment:
Could we just parametrize these two test (`@pytest.mark.parametrize('test_train_step_end', [False, True]`)?
Initialize model with
``` python
    class TestModel(BoringModel):
        def training_step(self, batch, batch_idx):
            return {"train_loss": 0}
        def training_step_end(self, out):
            d = {"train_loss": out["train_loss"]}
            return d
```
and then override depending on test arg:
```
model.training_step_end = model.training_step_end if test_train_step_end else None
```